src/backend/cpu/TNJ/BinaryNode.hpp

```diff
@@ -44,14 +44,24 @@ namespace TNJ
         {
         }
 
-        void calc(int x, int y, int z, int w)
+        void calc(int x, int y, int z, int w, int lim)
         {
-            this->m_val = m_op.eval(m_lhs->m_val, m_rhs->m_val);
+            Ti *lhs_ptr = &(m_lhs->m_val.front());
```

Comment:
`std::array` also have `data` member function similar to `std::vector`. You can use that directly instead of doing & on output of front call.